x/stakeibc/keeper/transfer.go

```diff
@@ -0,0 +1,146 @@
+
+import (
+	"errors"
+	"fmt"
+
+	errorsmod "cosmossdk.io/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/cosmos/gogoproto/proto"
+	transfertypes "github.com/cosmos/ibc-go/v7/modules/apps/transfer/types"
+	clienttypes "github.com/cosmos/ibc-go/v7/modules/core/02-client/types"
+	channeltypes "github.com/cosmos/ibc-go/v7/modules/core/04-channel/types"
+
+	epochstypes "github.com/Stride-Labs/stride/v14/x/epochs/types"
+	"github.com/Stride-Labs/stride/v14/x/stakeibc/types"
+)
+
+// Transfers tokens from the community pool deposit ICA account to the host zone stake holding module address for that pool
+func (k Keeper) TransferCommunityPoolDepositToHolding(ctx sdk.Context, hostZone types.HostZone, token sdk.Coin) error {
+	// Verify that the deposit ica address exists on the host zone and stake holding address exists on stride
+	if hostZone.CommunityPoolDepositIcaAddress == "" || hostZone.CommunityPoolStakeHoldingAddress == "" {
+		return errors.New("Invalid deposit address or stake holding address, cannot build valid ICA transfer kickoff command")
+	}
+
+	// get the hostZone counterparty transfer channel for sending tokens from hostZone to Stride
+	transferChannel, found := k.IBCKeeper.ChannelKeeper.GetChannel(ctx, transfertypes.PortID, hostZone.TransferChannelId)
+	if !found {
+		return errorsmod.Wrapf(channeltypes.ErrChannelNotFound, "transfer channel %s not found", hostZone.TransferChannelId)
+	}
+	counterpartyChannelId := transferChannel.Counterparty.ChannelId
+
+	// Timeout both the ICA kick off command and the ibc transfer message at the epoch boundary
+	strideEpochTracker, found := k.GetEpochTracker(ctx, epochstypes.STRIDE_EPOCH)
+	if !found {
+		return errorsmod.Wrapf(types.ErrEpochNotFound, epochstypes.STRIDE_EPOCH)
+	}
+	endEpochTimestamp := uint64(strideEpochTracker.NextEpochStartTime)
+
+	// Determine the host zone's stToken ibc denom
+	nativeDenom := hostZone.HostDenom
+	stIbcDenom, err := k.GetStIbcDenomOnHostZone(ctx, hostZone)
+	if err != nil {
+		return err
+	}
+
+	// If the token is an stToken, we send it to the redeem holding address to be redeemed
+	// Otherwise, we send it to the stake holding address to be liquid staked
+	var destinationHoldingAddress string
+	switch token.Denom {
+	case nativeDenom:
+		destinationHoldingAddress = hostZone.CommunityPoolStakeHoldingAddress
+	case stIbcDenom:
+		destinationHoldingAddress = hostZone.CommunityPoolRedeemHoldingAddress
+	default:
+		return fmt.Errorf("Invalid community pool transfer denom: %s", token.Denom)
+	}
+
+	memo := ""
+	var msgs []proto.Message
+	msgs = append(msgs, transfertypes.NewMsgTransfer(
+		transfertypes.PortID,
+		counterpartyChannelId, // for transfers of communityPoolHostZone -> Stride
+		token,
+		hostZone.CommunityPoolDepositIcaAddress, // ICA controlled address on community pool zone
+		destinationHoldingAddress,               // Stride address, unique to each community pool / hostzone
+		clienttypes.Height{},
+		endEpochTimestamp,
```

Comment:
Does this msg type not need both timeoutHeight and timeoutTimestamp — it looks like it's just getting a Timestamp only.

Reply:
only needs one of either height or timestamp!